vnext/src/Libraries/Image/resolveAssetSource.windows.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+// Resolves an asset into a `source` for `Image`.
+
+'use strict';
+
+const AssetRegistry = require('@react-native/assets/registry');
+const AssetSourceResolver = require('./AssetSourceResolver');
+const Platform = require('../Utilities/Platform');
+const {pickScale} = require('./AssetUtils');
+
+import type {ResolvedAssetSource} from './AssetSourceResolver';
+
+let _customSourceTransformer, _serverURL, _scriptURL;
+
+let _sourceCodeScriptURL: ?string;
+function getSourceCodeScriptURL(): ?string {
+  if (_sourceCodeScriptURL) {
+    return _sourceCodeScriptURL;
+  }
+
+  let sourceCode =
+    global.nativeExtensions && global.nativeExtensions.SourceCode;
+  if (!sourceCode) {
+    sourceCode = require('../NativeModules/specs/NativeSourceCode').default;
+  }
+  _sourceCodeScriptURL = sourceCode.getConstants().scriptURL;
+  return _sourceCodeScriptURL;
+}
+
+function getDevServerURL(): ?string {
+  if (_serverURL === undefined) {
+    const sourceCodeScriptURL = getSourceCodeScriptURL();
+    const match =
+      sourceCodeScriptURL && sourceCodeScriptURL.match(/^https?:\/\/.*?\//);
+    if (match) {
+      // jsBundle was loaded from network
+      _serverURL = match[0];
+    } else {
+      // jsBundle was loaded from file
+      _serverURL = null;
+    }
+  }
+  return _serverURL;
+}
+
+function _coerceLocalScriptURL(scriptURL: ?string): ?string {
+  if (Platform.OS === 'windows') {
+    return scriptUrl;
+  }
+
+  if (scriptURL) {
+    if (scriptURL.startsWith('assets://')) {
+      // android: running from within assets, no offline path to use
+      return null;
+    }
+    scriptURL = scriptURL.substring(0, scriptURL.lastIndexOf('/') + 1);
+    if (!scriptURL.includes('://')) {
+      // Add file protocol in case we have an absolute file path and not a URL.
+      // This shouldn't really be necessary. scriptURL should be a URL.
+      scriptURL = 'file://' + scriptURL;
+    }
+  }
+  return scriptURL;
+}
+
+function getScriptURL(): ?string {
+  if (_scriptURL === undefined) {
```

Comment:
I can't see where _scriptURL comes from, but under which conditions would it be undefined? note that for image urls we support several non x-plat schemes apart from http/https, like ms-appx, ms-resource, resource, and some others; want to make sure this doesn't force these urls to be only "http/https" or "file" schemed

Reply:
It's just memoizing a result.

This change should be independent of the actual bundle root path, whether it's `ms-appx://`, `ms-resource://` or `C:\...` for unpackaged apps.

Reply:
Here is where the memoization variable is declared: https://github.com/microsoft/react-native-windows/pull/10621/files/41a3c9a59bf619cbd37e6eae96c607cea1ef4552#diff-7ef60247116c017061f149892f29f5977c15debf02f5bac3038ba60e7c85b99aR22